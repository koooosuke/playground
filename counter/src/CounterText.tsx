import React, { FC } from 'react';
import { View, Text, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  textContainer: {
    width: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 20,
  },
});

interface Props {
  count: number;
}

const CounterText: FC<Props> = ({ count }) => (
  <View style={styles.textContainer}>
    {count >= 10 && <Text>you&apos;re great!</Text>}
    <Text style={styles.text}>{count}</Text>
  </View>
);

export default CounterText;
