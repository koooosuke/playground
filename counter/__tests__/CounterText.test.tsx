import React from 'react';
import renderer from 'react-test-renderer';
import CounterText from '../src/CounterText';

describe('CounterText', () => {
  it('通常のCounterTextを描画できる', () => {
    const component = renderer.create(<CounterText count={0} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
