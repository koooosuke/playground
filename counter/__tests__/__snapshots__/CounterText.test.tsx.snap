// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterText 通常のCounterTextを描画できる 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    0
  </Text>
</View>
`;
